pkg/southbound/synchronizer/synchronizer.go

```diff
@@ -118,45 +123,55 @@ func New(context context.Context, changeStore *store.ChangeStore, configStore *s
 
 // syncConfigEventsToDevice is a go routine that listens out for configuration events specific
 // to a device and propagates them downwards through southbound interface
-func (sync *Synchronizer) syncConfigEventsToDevice() {
+func (sync *Synchronizer) syncConfigEventsToDevice(errChan chan<- events.ErrorEvent) {
 
 	for deviceConfigEvent := range sync.deviceConfigChan {
-		change := sync.ChangeStore.Store[deviceConfigEvent.ChangeID()]
-		err := change.IsValid()
+		c := sync.ChangeStore.Store[deviceConfigEvent.ChangeID()]
+		err := c.IsValid()
 		if err != nil {
 			log.Warning("Event discarded because change is invalid ", err)
+			errChan <- events.CreateErrorEvent(events.EventTypeErrorParseConfig,
+				sync.key.DeviceID, c.ID, err)
 			continue
 		}
-		gnmiChange, parseError := change.GnmiChange()
+		gnmiChange, parseError := c.GnmiChange()
 
 		if parseError != nil {
 			log.Error("Parsing error for Gnmi change ", parseError)
+			errChan <- events.CreateErrorEvent(events.EventTypeErrorParseConfig,
+				sync.key.DeviceID, c.ID, err)
 			continue
 		}
 
 		log.Info("Change formatted to gNMI setRequest ", gnmiChange)
 		target, err := southbound.GetTarget(sync.key)
 		if err != nil {
 			log.Warning(err)
+			errChan <- events.CreateErrorEvent(events.EventTypeErrorDeviceConnect,
+				sync.key.DeviceID, c.ID, err)
 			continue
 		}
 		setResponse, err := target.Set(sync.Context, gnmiChange)
 		if err != nil {
-			log.Error("SetResponse ", err)
+			log.Error("Error while doing set ", err)
+			errChan <- events.CreateErrorEvent(events.EventTypeErrorSetConfig,
+				sync.key.DeviceID, c.ID, err)
 			continue
 		}
 		log.Info(sync.Device.Addr, " SetResponse ", setResponse)
 
 	}
 }
 
-func (sync Synchronizer) syncOperationalState() error {
+func (sync Synchronizer) syncOperationalState(errChan chan<- events.ErrorEvent) {
 
 	target, err := southbound.GetTarget(sync.key)
 
 	if err != nil {
 		log.Error("Can't find target for key ", sync.key)
-		return err
+		errChan <- events.CreateErrorEvent(events.EventTypeErrorDeviceConnect,
+			sync.key.DeviceID, make([]byte, 0), err)
+		return
```

Comment:
this means that this go routine is dies at this point? Probably OK.

Reply:
yes